src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -672,23 +672,13 @@ public override void DidCompleteWithError (NSUrlSession session, NSUrlSessionTas
 
 			void SetResponse (InflightData inflight)
 			{
-				lock (inflight.Lock) {
-					if (inflight.ResponseSent)
-						return;
-
-					if (inflight.CancellationTokenSource.Token.IsCancellationRequested)
-						return;
-
-					if (inflight.CompletionSource.Task.IsCompleted)
-						return;
-
-					var httpResponse = inflight.Response;
+				if (inflight.CancellationTokenSource.Token.IsCancellationRequested)
+					return;
 
-					inflight.ResponseSent = true;
+				if (inflight.CompletionSource.Task.IsCompleted)
+					return;
```

Comment:
This is not needed, `TrySetResult` will check for this as well (in a thread-safe manner).